include/glaze/ext/jsonrpc.hpp

```diff
@@ -0,0 +1,548 @@
+   struct server
+   {
+      server() = default;
+      glz::tuplet::tuple<method_type...> methods{};
+
+      template <detail::basic_fixed_string name>
+      constexpr void on(auto const& callback)  // std::function<expected<result_t, rpc::error>(params_t const&)>
+      {
+         detail::set_callback<name>(methods, callback);
+      }
+
+      // Return json stringified response_t meaning it can both be error and non-error
+      // If `id` is null in json_request a response will not be generated
+      std::vector<std::pair<std::string, rpc::error>> call(std::string_view json_request)
+      {
+         if (auto parse_err{glz::validate_json(json_request)}) {
+            generic_response_t response_parse_error{
+               rpc::error(rpc::error_e::parse_error, format_error(parse_err, json_request))};
+            return {{glz::write_json(response_parse_error), response_parse_error.error.value()}};
+         }
+
+         // Extract each request, if request does not include and ID a std::nullopt will be returned
+         auto const per_request{
+            [this](std::string_view json_request) -> std::optional<std::pair<std::string, rpc::error>> {
+               expected<generic_request_t, glz::parse_error> request{glz::read_json<generic_request_t>(json_request)};
+
+               if (!request.has_value()) {
+                  // Failed, but let's try to extract the `id`
+                  expected<json_t, glz::parse_error> exp_json_parsed{glz::read_json<json_t>(json_request)};
+                  if (exp_json_parsed.has_value()) {
+                     auto const& json_parsed{exp_json_parsed.value()};
+                     if (json_parsed.contains("id")) {
+                        generic_response_t response_w_id{json_parsed["id"].as<jsonrpc_id_type>(),
+                                                         rpc::error::invalid(request.error(), json_request)};
+                        return std::make_pair(glz::write_json(response_w_id), response_w_id.error.value());
+                     }
+                  }
+                  generic_response_t response_no_id{rpc::error::invalid(request.error(), json_request)};
+                  return std::make_pair(glz::write_json(response_no_id), response_no_id.error.value());
+               }
+
+               auto& req{request.value()};
+               if (req.version != rpc::supported_version) {
+                  if (std::holds_alternative<glz::json_t::null_t>(req.id)) {
+                     return std::nullopt;
+                  }
+                  generic_response_t response_inv_version{std::move(req.id), rpc::error::version(req.version)};
+                  return std::make_pair(glz::write_json(response_inv_version), response_inv_version.error.value());
+               }
+
+               std::optional<std::pair<std::string, rpc::error>> return_v{};
+               bool method_found = methods.any([&json_request, &req, &return_v](auto&& method) -> bool {
```

Comment:
yes, this is runtime comparison so I am not sure if this could be optimized further other then some hash map magic which you guys implemented for objects. But if the user would place the most common methods at the start it should be pretty performant. If it would require small changes to make this like objects we should definitely consider it.

Reply:
That can always be visited later and likely requires a bunch of methods to be worthwhile.